pkg/steps/multi_stage.go

```diff
@@ -255,6 +258,33 @@ func (s *multiStageTestStep) createSecret() error {
 	return err
 }
 
+func (s *multiStageTestStep) createCredentials() error {
+	log.Printf("Creating multi-stage test credentials for %q", s.name)
+	for _, step := range append(s.pre, append(s.test, s.post...)...) {
+		for _, credential := range step.Credentials {
```

Comment:
We'll likely have cases where more than one step in a test will use the same secret, so we should gather and deduplicate them before this loop.

Reply:
Also, I had the option to declare credentials at higher levels of the hierarchy (chains, workflows) in the original design document, so we wouldn't need to repeat them in all steps if all of them needed a certain credential.

That can be done in a separate PR, if we think it's a good idea. Even then the implementation here would not change.